### PR TITLE
Added link checks for .md files to CI

### PR DESCRIPTION
**Pull request by PurpleBriar**
# Type of PR

- [ ] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
Added link checks to .md files to check for dead links. Updated dead links in .md files

## Validation

- [ ] Unit tests updated and ran successfully
- [x] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes #issue_number (this will automatically close the issue when the PR closes)
- References #helium/issues/542

### CODE REVIEW

File: .github/workflows/mdlinkcheck.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        use-quiet-mode: 'yes'
+        use-verbose-mode: 'yes'
+        config-file: 'mlc_config.json'
```
Comment (sivamu) [inline .github/workflows/mdlinkcheck.yml:14]: Need a new line at end, also for your json file.

File: mlc_config.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+	"ignorePatterns": [
+		{
+			"pattern": "https://github.com/\\["
```
Comment (sivamu) [inline mlc_config.json:4]: Not sure whether the github based links should be ignored.

Comment (PurpleBriar) [reply mlc_config.json:4]: This only ignores the ones that have the "[" character after github.com, such as _**https://github.com/[link-name]**_. There were a few of these and they were incorrectly identified as dead links

Comment (sivamu) [reply mlc_config.json:4]: Ah gotcha, thanks for the clarification.

File: .github/workflows/mdlinkcheck.yml (hunk 1 above)
Comment (PurpleBriar) [reply .github/workflows/mdlinkcheck.yml]: Added new lines to both files.

**Review by sivamu — APPROVED** [review]

**Pull request merged**